editors/code/src/extension.ts

```diff
@@ -181,11 +181,16 @@ const startLanguageServer = async (context: vscode.ExtensionContext) => {
     }
   }
 
-  const serverBinPath = vscode.Uri.joinPath(
-    context.extensionUri,
-    "bin",
-    os.platform() === "win32" ? "server.exe" : "server",
+  const serverBinConfig = vscode.Uri.file(
+    vscode.workspace.getConfiguration("luau-lsp").get("serverPath", ""),
   ).fsPath;
+  const serverBinPath = serverBinConfig
+    ? serverBinConfig
```

Comment:
Calling `vscode.Uri.file` with an empty string, and then relying on fsPath being undefined, seems odd to me. Maybe we should do something like below:

```suggestion
  const serverBinConfig = vscode.workspace.getConfiguration("luau-lsp").get("serverPath");
  const serverBinPath = serverBinConfig
    ? vscode.Uri.file(serverBinConfig).fsPath
```

Also, similar to how we load definitions files, it might make sense to do an `exists` check on the provided binary to check if it is actually there, and emit an error if it isn't.

Reply:
I'll take a look at doing that `exists` check, thanks for the review!